### PR TITLE
KFSPTS-19974: Update to explicitly use Locale.US for formatting operations.

### PR DESCRIPTION
**Pull request by nkimble**

### CODE REVIEW

**Comment by nkimble** [comment]
This PR is ready for code review but should not be merged until we coordinate an appropriate time with the functional users.

**Comment by nkimble** [comment]
Reopening this PR to have files reviewed with any issues addressed now.
Please do not merge this PR until after the 12/17 upgrade patch release is merged to develop first.

File: src/main/java/edu/cornell/kfs/sys/xmladapters/StringToJavaDateAdapter.java
Comment (jhulslander) [inline src/main/java/edu/cornell/kfs/sys/xmladapters/StringToJavaDateAdapter.java:20]: can the format string be moved to a new constant in CUKFSConstants

File: src/test/java/edu/cornell/kfs/fp/batch/service/impl/CreateAccountingDocumentServiceImplTest.java
Comment (jhulslander) [inline src/test/java/edu/cornell/kfs/fp/batch/service/impl/CreateAccountingDocumentServiceImplTest.java:146]: can the format string be moved to a new constant in CUKFSConstants

File: src/test/java/edu/cornell/kfs/pdp/batch/service/impl/CuExtractPaymentServiceImplTest.java
Comment (jhulslander) [inline src/test/java/edu/cornell/kfs/pdp/batch/service/impl/CuExtractPaymentServiceImplTest.java:48]: can the format string be moved to a new constant in CUKFSConstants

**Review by jhulslander — CHANGES_REQUESTED** [review]
Changes look very good overall, I found some commented out variables I think can be removed, and a few cases where some format strings could be moved to new constants in CUKFSConstants

File: src/main/java/com/rsmart/kuali/kfs/cr/businessobject/CheckReconciliationReport.java
Comment (nkimble) [inline src/main/java/com/rsmart/kuali/kfs/cr/businessobject/CheckReconciliationReport.java:48]: Added a constant for this string as well.

File: src/main/java/com/rsmart/kuali/kfs/fp/batch/DisbursementVoucherInputFileType.java
Comment (nkimble) [reply src/main/java/com/rsmart/kuali/kfs/fp/batch/DisbursementVoucherInputFileType.java:49]: Added constant

File: src/main/java/edu/cornell/kfs/fp/batch/service/impl/AdvanceDepositServiceImpl.java
Comment (nkimble) [reply src/main/java/edu/cornell/kfs/fp/batch/service/impl/AdvanceDepositServiceImpl.java:292]: Constant used for the format string.

File: src/main/java/edu/cornell/kfs/module/purap/batch/IWantDocumentInputFileType.java
Comment (nkimble) [reply src/main/java/edu/cornell/kfs/module/purap/batch/IWantDocumentInputFileType.java:27]: Changed to use a constant.

File: src/main/java/edu/cornell/kfs/module/receiptProcessing/service/impl/ReceiptProcessingServiceImpl.java
Comment (nkimble) [reply src/main/java/edu/cornell/kfs/module/receiptProcessing/service/impl/ReceiptProcessingServiceImpl.java:490]: Constant created and used.

File: src/main/java/edu/cornell/kfs/sys/batch/CuAbstractStep.java
Comment (nkimble) [reply src/main/java/edu/cornell/kfs/sys/batch/CuAbstractStep.java:43]: Constant created and used.

File: src/main/java/edu/cornell/kfs/sys/xmladapters/StringToJavaDateAdapter.java
Comment (nkimble) [reply src/main/java/edu/cornell/kfs/sys/xmladapters/StringToJavaDateAdapter.java:20]: Constant created.

File: src/test/java/edu/cornell/kfs/fp/batch/service/impl/CreateAccountingDocumentServiceImplTest.java
Comment (nkimble) [reply src/test/java/edu/cornell/kfs/fp/batch/service/impl/CreateAccountingDocumentServiceImplTest.java:146]: Constant created and used.

File: src/test/java/edu/cornell/kfs/pdp/batch/service/impl/CuExtractPaymentServiceImplTest.java
Comment (nkimble) [reply src/test/java/edu/cornell/kfs/pdp/batch/service/impl/CuExtractPaymentServiceImplTest.java:48]: Constant used.

**Comment by nkimble** [comment]
Unit tests are running. This is ready to be reviewed again.

File: src/main/java/edu/cornell/kfs/module/ld/batch/service/impl/CuFileEnterpriseFeederHelperServiceImpl.java
Comment (jhulslander) [inline src/main/java/edu/cornell/kfs/module/ld/batch/service/impl/CuFileEnterpriseFeederHelperServiceImpl.java:110]: Did you intentially switch from stream to parallelStream?

File: src/main/java/edu/cornell/kfs/sys/CUKFSConstants.java
Comment (jhulslander) [inline src/main/java/edu/cornell/kfs/sys/CUKFSConstants.java:30]: I think this is the same as KFSConstants.MONTH_DAY_YEAR_DATE_FORMAT

**Review by jhulslander — CHANGES_REQUESTED** [review]
Things are looking good, I had just a couple of questions.

File: src/main/java/edu/cornell/kfs/sys/CUKFSConstants.java
Comment (nkimble) [reply src/main/java/edu/cornell/kfs/sys/CUKFSConstants.java:30]: Yes it is. Adjusting for that change.

File: src/main/java/edu/cornell/kfs/module/ld/batch/service/impl/CuFileEnterpriseFeederHelperServiceImpl.java
Comment (nkimble) [reply src/main/java/edu/cornell/kfs/module/ld/batch/service/impl/CuFileEnterpriseFeederHelperServiceImpl.java:110]: Thanks for catching.

**Comment by jhulslander** [comment]
retest this please

**Review by jhulslander — APPROVED** [review]
my concerns have been addressed, unit tests pass, I shall merge

**Pull request merged**
